src/openapi/streaming/streaming.js

```diff
@@ -523,6 +525,83 @@ Streaming.prototype.disposeSubscription = function(subscription) {
     }
 };
 
+function createHandleSubscriptionReadyForUnsubscribe(subscriptions, resolve) {
+    return function() {
+        if (subscriptions.every((subscription) => subscription.isReadyForUnsubscribeByTag())) {
+            resolve();
+        }
+    };
+}
+
+function unsubscribeSubscriptionByTag(serviceGroup, url, tag, shouldDisposeSubscription) {
+    const subscriptionsToRemove = this.subscriptions.filter((subscription) =>
+        subscription.serviceGroup === serviceGroup &&
+            subscription.url === url &&
+            subscription.subscriptionData.Tag === tag
+        );
+
+    let allSubscriptionsReadyResolve;
+    const allSubscriptionsReady = new Promise((resolve) => {
+        allSubscriptionsReadyResolve = resolve;
+    });
+    const handleSubscriptionReadyForUnsubscribe = createHandleSubscriptionReadyForUnsubscribe(subscriptionsToRemove, allSubscriptionsReadyResolve);
+
+    subscriptionsToRemove.forEach((subscription) => {
+        ignoreSubscriptions[subscription.referenceId] = true;
+        setTimeout(() => delete ignoreSubscriptions[subscription.referenceId], MS_TO_IGNORE_DATA_ON_UNSUBSCRIBED);
+
+        subscription.addStateChangedCallback(handleSubscriptionReadyForUnsubscribe);
+        subscription.onUnsubscribeByTagPending();
+
+        if (shouldDisposeSubscription) {
+            subscription.dispose();
+
+            const indexOfSubscription = this.subscriptions.indexOf(subscription);
+            if (indexOfSubscription >= 0) {
+                this.subscriptions.splice(indexOfSubscription, 1);
+            }
+        }
+    });
+
+    allSubscriptionsReady.then(() => {
+        this.transport.delete(serviceGroup, url + '/{contextId}/?Tag={tag}', {
+            contextId: this.contextId,
+            tag,
+        })
+        .then(() => subscriptionsToRemove.forEach((subscription) => {
+            subscription.onUnsubscribeByTagComplete();
+            subscription.removeStateChangedCallback(handleSubscriptionReadyForUnsubscribe);
+        }))
+        .catch(() => subscriptionsToRemove.forEach((subscription) => {
+            subscription.onUnsubscribeByTagComplete();

```

Comment:
do we normally not do anything different if a unsubscribe call fails?
if we do keep this, do

.catch(noop)
.then(

to avoid duplicated code